### PR TITLE
Adds support for Coherence Topics and messaging

### PR DESCRIPTION
**Pull request by vasac**

### CODE REVIEW

**Review by thegridman — APPROVED** [review]
This looks ok. It will need changes to support committing messages when we have worked out how that will work.

**Comment by ghillert** [comment]
@vasac Please rebase and unless there is some good reason, squash the commits into one.

File: coherence-spring-core/src/main/java/com/oracle/coherence/spring/annotation/SubscriberGroup.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	/**
+	 * The name of the subscriber group.
+	 * @return the name of the subscriber group
+	 */
+	String value();
+
+	/**
+	 * An annotation literal for the {@link SubscriberGroup} annotation.
+	 */
+	class Literal extends AnnotationLiteral<SubscriberGroup> implements SubscriberGroup {
```
Comment (ghillert) [inline coherence-spring-core/src/main/java/com/oracle/coherence/spring/annotation/SubscriberGroup.java:37]: You can probably remove this whole section. I don't think we use it for anything. That will also get rid of your CheckStyle error

**Comment by ghillert** [comment]
Once you get past the CheckStyle error ... Are you able to build the project? I am seeing some strange:

```
Caused by: java.lang.ClassNotFoundException: org.springframework.jdbc.core.ConnectionCallback
 	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
 	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
 ```

**Comment by vasac** [comment]
> Once you get past the CheckStyle error ... Are you able to build the project?
I can build it just fine with maven wrapper while it's failing with my local maven - they pull in different Java versions.

**Comment by vasac** [comment]
@ghillert Classpath scanning for @CoherencePublisher annotated interface was rewritten, Spring Boot autoconfiguration was added and docs are updated to cover changes.

**Review by ghillert — APPROVED** [review]

**Comment by ghillert** [comment]
Rebased + merged with commit e3e459bb5583f4c46849b177d14084fc27da5af2

**Pull request closed without merging**